hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaObserverChore.java

```diff
@@ -433,6 +433,11 @@ void updateNamespaceQuota(
             LOG.info(tableInNS + " moving into violation of namespace space quota with policy "
                 + targetStatus.getPolicy());
             this.snapshotNotifier.transitionTable(tableInNS, targetSnapshot);
+            // when the Namespace is in violation due to Disable Policy, Disable the table
+            if (targetStatus.isInViolation()
```

Comment:
snapshotNotifier.transitionTable may throw IOE.
Do you want to enclose the call to snapshotNotifier.transitionTable in try / catch block so that the new code would always run ?

Reply:
hi @tedyu yes I can enclose it in a try/catch but I have a small question. If the `snapshotNotifier.transitionTable` failed with an exception and if it is caught and the table is disabled, wouldn't there be inconsistency as the quota table will still show the table as non-violated(wrong usage and limit) but at the same time the table will be disabled because of Quota violation?

Reply:
The else block starting at line 418 is for targetStatus.isInViolation()
So if SpaceViolationPolicy.DISABLE is in effect, I think the table should be disabled.

We'd better keep quota table consistent with regard to the violation.

Reply:
Yes, that makes sense. I have updated the PR.